code/modules/research/nanites/nanite_cloud_controller.dm

```diff
@@ -239,9 +244,51 @@
 				var/datum/nanite_program/P = nanites.programs[text2num(params["program_id"])]
 				var/datum/nanite_rule/rule = P.rules[text2num(params["rule_id"])]
 				rule.remove()
+				balloon_alert(usr, "removed rule")
 
 				investigate_log("[key_name(usr)] removed rule [rule.display()] from program [P.name] in cloud #[current_view]", INVESTIGATE_NANITES)
 				. = TRUE
+		if("combine_rules")
+			var/datum/nanite_cloud_backup/backup = get_backup(current_view)
+			if(backup)
+				playsound(src, 'sound/machines/terminal_prompt.ogg', 50, 0)
+				var/datum/component/nanites/nanites = backup.nanites
+				var/datum/nanite_program/program = nanites.programs[text2num(params["program_id"])]
+				if(!islist(params["rule_ids"]))
+					return
+				if(length(params["rule_ids"]) <= 1)
+					to_chat(usr, "<span class='warning'>Warning: you need to combine at least 2 rules.</span>")
+					balloon_alert(usr, "not enough rules to combine")
+					return
+				if(length(params["rule_ids"]) > 5)
+					to_chat(usr, "<span class='warning'>Warning: cannot combine more than 5 rules.</span>")
+					balloon_alert(usr, "too many rules to combine")
+					return
+				var/list/datum/nanite_rule/rules_to_copy = list()
+				var/list/rule_displays = list()
+				for(var/rule_id in params["rule_ids"])
+					var/datum/nanite_rule/rule = program.rules[text2num(rule_id)]
+					if(!rule)
+						continue
+					if(!rule.combinable)
+						balloon_alert(usr, "rules cannot be combined")
+						to_chat(usr, "<span class='warning'>Warning: rule '[rule.display()]' cannot be combined.</span>")
+						return
+					rule_displays += rule.display()
+					rules_to_copy += rule
+				if(!length(rules_to_copy))
+					balloon_alert(usr, "no rules to combine")
+					to_chat(usr, "<span class='warning'>Warning: no rules to combine!</span>")
+					return
+				var/list/datum/nanite_rule/copied_rules = list()
+				for(var/R in rules_to_copy)
+					var/datum/nanite_rule/rule = R
+					copied_rules += rule.copy_to(program, FALSE)
+					rule.remove()
+				new /datum/nanite_rule/combined(program, TRUE, copied_rules, params["op"])
```

Comment:
Sanitise op before sending it into a function

Reply:
`/datum/nanite_rule/combined/New` sanitizes it anyways:
```dm
src.op = sanitize_inlist(op, NL_ALL, NL_AND)
```